### PR TITLE
test: refactor packages to jest

### PR DESCRIPTION
**Pull request by juanpicado**
/pnpm/packages/default-resolver @juanpicado
 /pnpm/packages/dependency-path @juanpicado
 /pnpm/packages/config @juanpicado

### CODE REVIEW

File: packages/config/test/index.ts [hunk 1]
```diff
@@ -180,40 +171,34 @@ test('when using --global, link-workspace-packages, shared-workspace-shrinwrap a
         version: '1.0.0',
       },
     })
-    t.notOk(config.linkWorkspacePackages, 'link-workspace-packages is false')
-    t.notOk(config.sharedWorkspaceLockfile, 'shared-workspace-lockfile is false')
-    t.notOk(config.lockfileDir, 'lockfile-dir is null')
+    expect(config.linkWorkspacePackages).toBeFalsy()
+    expect(config.sharedWorkspaceLockfile).toBeFalsy()
+    // FIXME: it supposed to return null but is undefined
+    expect(config.lockfileDir).toBeUndefined()
```
Comment (juanpicado) [inline packages/config/test/index.ts:177]: I've noticed here the comment expect `null` but the outcome is clearly `undefined`.

File: packages/config/test/index.ts [hunk 2]
```diff
@@ -462,14 +433,12 @@ test('normalizing the value of public-hoist-pattern', async (t) => {
...
     })
 
-    t.equal(config.publicHoistPattern, undefined)
+    expect(config.publicHoistPattern).toBeUndefined()
   }
-  t.end()
 })
 
-test('rawLocalConfig in a workspace', async (t) => {
+test('rawLocalConfig in a workspace', async () => {
```
Comment (juanpicado) [inline packages/config/test/index.ts]: This test fails locally if you have a `.npmrc` global file, the outcome does not match

Comment (juanpicado) [reply packages/config/test/index.ts]: Actually also fails here, maybe I did something wrong

Comment (zkochan) [reply packages/config/test/index.ts]: Let's skip the failing test. We'll need to refactor the config reader.

Comment (juanpicado) [reply packages/config/test/index.ts]: Ok, I'll add a comment and skip those.

**Review by zkochan — APPROVED** [review]

**Pull request merged**